### PR TITLE
#32968 fixed issue when order status history accepts wrong order status

### LINKED ISSUE #32968: REST orders/X/comments allows undefined statuses
When updating the order-status/comments using the REST API, it allows for non configured status-es to be set

### Preconditions (*)

1. Magento CE 2.3.4
2. PHP 7.3.X

### Steps to reproduce (*)

1. Update status/add comment to order using REST API url: /rest/V1/orders/[ORDERID]/comments
2. In the payload add a 'status'-string that is not configured in Magento

### Expected result (*)

Either:
- Deny the update, returning false instead of true or provide a descriptive error
- Show the unconfigured status on '/admin/sales/order/view/order_id/[ORDERID]'

### Actual result (*)

1. The rest route simply returns true, the data is stored
2. The '/admin/sales/order/view/order_id/[ORDERID]' show the update, but without any 'status' text, see screenshot:

Schermafbeelding 2021-05-07 om 10 13 36

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

Order statuses seem pretty connected to order states. So far I have no reason to see this as 'critical data', but I also haven't checked in depth if anything else breaks.

### PR DESCRIPTION
**Pull request by vovsky**
### Description (*)
This PR fixes issue when order status history accepts wrong order status

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32968

### Manual testing scenarios (*)

1. Use REST orders/X/comments endpoint to add Status History with wrong order status, e. g.
```
{
    "statusHistory": {
        "comment": "some comment",
        "created_at": "2021-05-26  15:22:04",
        "parent_id": 12,
        "is_customer_notified": 0,
        "is_visible_on_front": 1,
        "status": "wrong_status"
    }
}
```
OR 
1. in Admin edit Add comment Dropdown, add wrong Order status and submit comment

### CODE REVIEW

**Comment by vovsky** [comment]
@magento run all tests

**Comment by vovsky** [comment]
@magento run all tests

**Comment by vovsky** [comment]
@Den4ik updated. Thank you

**Comment by vovsky** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/Order.php [hunk 1]
```diff
@@ -393,6 +399,7 @@ public function __construct(
         \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $salesOrderCollectionFactory,
         PriceCurrencyInterface $priceCurrency,
         \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productListFactory,
+        OrderStateResolverInterface $orderStateResolver,
```
Comment (Den4ik) [inline app/code/Magento/Sales/Model/Order.php]: Please follow Backward Compatibility for adding new constructor params.
https://devdocs.magento.com/contributor-guide/backward-compatible-development/#php

**Review by Den4ik — CHANGES_REQUESTED** [review]
@vovsky Thanks for your work. Please look at my comment

File: app/code/Magento/Sales/Model/Order.php (hunk 1 above)
Comment (vovsky) [reply app/code/Magento/Sales/Model/Order.php]: done. Thank you

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@vovsky Thanks for update. Could you look at failed tests

**Comment by vovsky** [comment]
@magento run all tests

**Comment by vovsky** [comment]
@magento run all tests

**Comment by vovsky** [comment]
@magento run all tests

**Comment by vovsky** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run WebAPI Tests, Functional Tests EE, Functional Tests B2B

**Review by Den4ik — APPROVED** [review]
✅ Changes looks good for me.
@vovsky Thanks for your work.

**Comment by Den4ik** [comment]
@magento run Functional Tests B2B

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9128](https://jira.corp.magento.com/browse/ENGCOM-9128) has been created to process this Pull Request

**Comment by vovsky** [comment]
Feel free to resolve conflicts. I'm going to work on this issue in near future. Thanks

**Comment by engcom-Charlie** [comment]
Hi @vovsky,

Thank you for your contribution!

Did you get a chance to look into the conflicts. In order to proceed further, the conflicts should get resolve as mentioned above. Closing this PR now, @vovsky once you are ready with PR, please feel free to reopen it or let us know about the same. We will be happy to reopen it.

Thank you!

**Pull request closed without merging**